runtime-common/loader.ts

```diff
@@ -149,13 +141,13 @@ export class Loader {
     this.urlMappings.set(new RealmPaths(from), to.href);
   }
 
-  static addRealmFetchOverride(realm: Realm) {
+  static registerRealm(realm: Realm) {
     let loader = Loader.getLoader();
-    loader.addRealmFetchOverride(realm);
+    loader.registerRealm(realm);
   }
 
-  addRealmFetchOverride(realm: Realm) {
-    this.realmFetchOverride.push(realm);
+  registerRealm(realm: Realm) {
+    this.realms.add(realm);
```

Comment:
For the loader it turns out it's not as simple as just dealing with URL's and callbacks for loading a URL. remember that our routes are sensitive to the accept header. Because of this the realm actually has multiple routers based on the accept header, since otherwise the routing rules would collide with one another. Otherwise the loader would not be able to tell the difference between a module fetch request and an instance fetch request. To make things simple, I left the logic around choosing the realm to use for the requested URL as-is, since the realm is able to figure out which callback to use based on both the accept header and the request URL

Reply:
this was really close, just pass a URL instead of the Realm